examples/plot_nonstationary_error.py

```diff
@@ -0,0 +1,213 @@
+
+df = xdem.spatialstats.nd_binning(values=dh.data, list_var=[slope, aspect, planc, profc],
+                                  list_var_names=['slope','aspect','planc','profc'],
+                                  statistics=['count', xdem.spatialstats.nmad],
+                                  list_var_bins=30)
+
+# %%
+# We obtain a dataframe with the 1D binning results for each variable, the 2D binning results for all combinations of
+# variables and the N-D (here 4D) binning with all variables.
+# Overview of the dataframe structure for the 1D binning:
+df[df.nd == 1]
+
+# %%
+# And for the 4D binning:
+df[df.nd == 4]
+
+# %%
+# We can now visualize the results of the 1D binning of the computed NMAD of elevation differences with each variable
+# using :func:`xdem.spatialstats.plot_1d_binning`.
+# We can start with the slope that has been long known to be related to the elevation measurement error (e.g.,
+# `Toutin (2002) <https://doi.org/10.1109/TGRS.2002.802878>`_).
+xdem.spatialstats.plot_1d_binning(df, 'slope', 'nmad', 'Slope (degrees)', 'NMAD of dh (m)')
+
+# %%
+# We identify a clear variability, with the dispersion estimated from the NMAD increasing from ~2 meters for nearly flat
+# slopes to above 12 meters for slopes steeper than 50°.
+# In statistical terms, such a variability of `variance <https://en.wikipedia.org/wiki/Variance>`_ is referred as
+# `heteroscedasticity <https://en.wikipedia.org/wiki/Heteroscedasticity>`_. Here we observe heteroscedastic elevation
+# differences due to a non-stationarity of variance with the terrain slope.
+#
+# What about the aspect?
+
+xdem.spatialstats.plot_1d_binning(df, 'aspect', 'nmad', 'Aspect (degrees)', 'NMAD of dh (m)')
+
+# %%
+# There is no variability with the aspect which shows a dispersion averaging 2-3 meters, i.e. that of the complete sample.
+#
+# What about the plan curvature?
+
+xdem.spatialstats.plot_1d_binning(df, 'planc', 'nmad', 'Planform curvature (100 m$^{-1}$)', 'NMAD of dh (m)')
+
+# %%
+# The relation with the plan curvature remains ambiguous.
+# We should better define our bins to avoid sampling bins with too many or too few samples. For this, we can partition
+# the data in quantiles in :func:`xdem.spatialstats.nd_binning`.
+# Note: we need a higher number of bins to work with quantiles and still resolve the edges of the distribution. Thus, as
+# with many dimensions the N dimensional bin size increases exponentially, we avoid binning all variables at the same
+# time and instead bin one at a time.
+# We define 1000 quantile bins of size 0.001 (equivalent to 0.1% percentile bins) for the profile curvature:
+
+df = xdem.spatialstats.nd_binning(values=dh.data, list_var=[profc], list_var_names=['profc'],
+                                  statistics=['count', np.nanmedian, xdem.spatialstats.nmad],
+                                  list_var_bins=[[np.nanquantile(profc,0.001*i) for i in range(1001)]])
```

Comment:
`np.nanquantile(profc, np.linspace(0, 1, 1000))` is slightly easier syntax!

Reply:
you're right, I changed all of those!

Reply:
in fa471c9